src/index.js

```diff
@@ -1,27 +1,30 @@
 const functions = require('@google-cloud/functions-framework')
 
-const currentDate = new Date().toISOString().substring(0, 10)
 const TRIGGERS = {
   cwv_tech_report: {
     type: 'poller',
     query: `
-SELECT LOGICAL_AND(condition)
-FROM (
-  SELECT TOTAL_ROWS > 0 AS condition
-  FROM \`chrome-ux-report.materialized.INFORMATION_SCHEMA.PARTITIONS\`
-  WHERE TABLE_NAME = 'device_summary'
-    AND PARTITION_ID = FORMAT_DATE('%Y%m%d', DATE_SUB(DATE_TRUNC(DATE '${currentDate}', MONTH), INTERVAL 1 MONTH))
-  UNION ALL
-  SELECT TOTAL_ROWS > 0 AS condition
-  FROM \`chrome-ux-report.materialized.INFORMATION_SCHEMA.PARTITIONS\`
-  WHERE TABLE_NAME = 'device_summary'
-    AND PARTITION_ID = FORMAT_DATE('%Y%m%d', DATE_SUB(DATE_TRUNC(DATE '${currentDate}', MONTH), INTERVAL 1 MONTH))
-  UNION ALL
-  SELECT NOT TOTAL_ROWS > 0 AS condition
-  FROM \`httparchive.core_web_vitals.INFORMATION_SCHEMA.PARTITIONS\`
-  WHERE TABLE_NAME = 'technologies'
-    AND PARTITION_ID = FORMAT_DATE('%Y%m%d', DATE_SUB(DATE_TRUNC(DATE '${currentDate}', MONTH), INTERVAL 1 MONTH))
-);
+DECLARE previousMonth STRING DEFAULT FORMAT_DATE('%Y%m%d', DATE_SUB(DATE_TRUNC(CURRENT_DATE(), MONTH), INTERVAL 1 MONTH));
+DECLARE previousMonth_YYYYMM STRING DEFAULT SUBSTR(previousMonth, 1, 6);
+
+WITH crux AS (
+  SELECT
+    LOGICAL_AND(total_rows > 20000000) AS min_rows_available,
```

Comment:
Ditto

Reply:
Reversed it to check for 'not empty rows` only. The rest of validation to done in Dataform asserts.